DbProvider.js

```diff
@@ -0,0 +1,59 @@
+import { useState, useEffect, createContext } from "react";
+import { Text, View } from "react-native";
+import * as SQLite from "expo-sqlite";
+
+export const DBProvider = createContext(undefined);
+
+export default function DataBaseProvider({ children }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [database, setDatabase] = useState(null);
+
+  useEffect(() => {
+    initialiseDatabase().then((db) => {
+      setIsLoading(false);
+      setDatabase(db);
+    });
+  }, []);
+
+  if (isLoading) {
+    return (
+      <View style={{ alignItems: "center", justifyContent: "center", flex: 1 }}>
+        <Text>Loading data from database :) ...</Text>
+      </View>
+    );
+  } else {
+    return (
+      <DBProvider.Provider value={database}>{children}</DBProvider.Provider>
+    );
+  }
+}
+
+async function initialiseDatabase() {
+  // open database?
+  const db = await SQLite.openDatabaseAsync("data");
+  // setup table for weight
+  await db.execAsync(`
+      CREATE TABLE IF NOT EXISTS weight (
+        time INTEGER NOT NULL,
+        date STRING NOT NULL,
```

Comment:
you should not add a new `date` column because you can infer the date from the time! You can use the the `Date` javascript library to do this, although you'll have to be careful with timezones (I would just stick to UTC for now)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date

Reply:
though I would also be tempted to never care about the date and only ever do INSERT rather than UPDATE. That way you would allow multiple readings per day but that's fine I think